### ISSUE TITLE
after training yolo world and use the trained model, model.set_classes() doesn't work properly

### ISSUE THREAD

**Issue by lwll** [ultralytics/ultralytics#18273]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I use COCO8 Dataset to train yolo world, like this:

```python
from ultralytics import YOLO

# Load a model
model = YOLO("../models/yolov8x-worldv2.pt")  # load a pretrained model (recommended for training)

# Train the model
results = model.train(data="../dataset/data.yaml", epochs=100, imgsz=640, device=1, workers=1)
```
and use the new model:
```python
from ultralytics import YOLO
model = YOLO("../models/best.pt")
model.set_classes(["hat", "backpack", "person", ""])
results = model("../sample/knife2.jpg", conf=0.05, iou=0.50, augment=True, agnostic_nms=True)
```
it seems, after traning the yolo world model,  model.set_classes() can only pass the class I trained before, and it must be the same order I trained.
Is there a way to still use the Open-Vocabulary Detection?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lwll, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it. A clear and minimal snippet showcasing the issue would be helpful in reproducing and resolving the problem on our end.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset details, model configurations, training logs, and any modifications made. This will help us better understand your needs. Additionally, verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

You mentioned leveraging `model.set_classes()` for Open-Vocabulary Detection. While this is not something we can directly resolve here, rest assured an Ultralytics engineer will review the issue and assist you soon 👨‍💻☕.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Is there an error? What's the actual issue?

**Comment by lwll** [comment]
> Is there an error? What's the actual issue?

no error. here is the result:
![17344312406694](https://github.com/user-attachments/assets/5b0c9bbe-3e69-4097-a1a7-cc6ec028ddfb)
and here is my custom classes and traned coco.yaml
![image](https://github.com/user-attachments/assets/da40d0d8-0918-4c08-8e04-39b09d570536)
you can see the result should be _backpack_ and _person_

**Comment by Y-T-G** [comment]
Use `YOLOWorld` to load the model and it should work.

```python
from ultralytics import YOLOWorld

model = YOLOWorld("best.pt")
```

**Comment by Y-T-G** [comment]
@Laughing-q It seems like the `YOLO()` doesn't automatically use `YOLOWorld()` after the model is fine-tuned.

**Comment by lwll** [comment]
it works now. thanks!

**Comment by Laughing-q** [comment]
> @Laughing-q It seems like the `YOLO()` doesn't automatically use `YOLOWorld()` after the model is fine-tuned.

oh yes we simply handle the logic there only when there's `world` in the name of the model. Using `YOLOWorld` for models without `world` as a part of the name works properly.
